scipy/stats/_distn_infrastructure.py

```diff
@@ -3237,6 +3237,9 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
             raise ValueError("xk and pk need to have the same length.")
         if not np.allclose(np.sum(pk), 1):
             raise ValueError("The sum of provided pk is not 1.")
+        if np.any(np.asarray(xk).astype(int) != xk):
+            raise ValueError("xk must be integers. If you really want floats, "
+                             "use xk as integer indices into the float array.")
```

Comment:
It's not immediately obvious where to put that float array. I'm assuming you mean in user code, so convert ``xk`` to ``floatarr[xk]`` for use with ``rv_sample.some_method(xk)``?